staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -27,6 +27,9 @@ var (
 	//   defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, <value>)()
 	DefaultMutableFeatureGate featuregate.MutableFeatureGate = featuregate.NewFeatureGate()
 
+	// DefaultMutableServiceGate is similar to DefaultMutableFeatureGate, but is intended for API server feature gates.
+	DefaultMutableServiceGate featuregate.MutableFeatureGate = featuregate.NewFeatureGate()
```

Comment:
sorry, I think we are talking past each other. This file is being used to set the process-wide feature gates. So we only need to keep `DefaultMutableFeatureGate`.

The e2e_node process can be configured with two arguments: one defines feature gates that will be passed to  the kubelet process that test will start. Another argument defines feature gates to set for the fake API server that will be served by the test.

The fact there are two different arguments for e2e_node is a specific of test configuration and should not leak into the common library like this.

Reply:
Thanks for the clarification. I've now moved `DefaultMutableServiceGate` to `test/e2e_node/util.go` and made it unexposed (i.e. "private").

Does this fulfills your expectations? If it doesn't, please kindly clarify your expectations.